crates/net/rpc/src/eth/signer.rs

```diff
@@ -0,0 +1,58 @@
+//! An abstraction over ethereum signers.
+
+use jsonrpsee::core::RpcResult as Result;
+use reth_primitives::{Address, Signature, TransactionSigned};
+use reth_rpc_types::TypedTransactionRequest;
+use secp256k1::SecretKey;
+use std::collections::HashMap;
+
+/// An Ethereum Signer used via RPC.
+#[async_trait::async_trait]
+pub(crate) trait EthSigner: Send + Sync {
+    /// Returns the available accounts for this signer.
+    fn accounts(&self) -> Vec<Address>;
+
+    /// Returns `true` whether this signer can sign for this address
+    fn is_signer_for(&self, addr: Address) -> bool {
```

Comment:
shall we accept addresses as ref? same for methods below 
```suggestion
    fn is_signer_for(&self, addr: &Address) -> bool {
```